controllers/job_test.go

```diff
@@ -691,3 +692,189 @@ func TestJobControllerPost_Error(t *testing.T) {
 	})
 }
+
+func TestJobControllerPost_Timeout_DifferentTime(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		current data.JobStatus
+		next    data.JobStatus
+		timeout uint
+	}{
+		{current: data.JobQueued, next: data.JobInflight, timeout: 10},
+		{current: data.JobQueued, next: data.JobInflight, timeout: 1000},
+		{current: data.JobQueued, next: data.JobInflight, timeout: 100000},
+	}
+	for _, test := range tests {
+		testName := "Success:202-Accepted Timeout Different Time " + strconv.FormatUint(uint64(test.timeout), 10) + " " + test.current.String() + " to " + test.next.String()
+		test := test
+		t.Run(testName, func(t *testing.T) {
+			t.Parallel()
+			job, err := setupTestJob()
+			assert.NoError(t, err)
+
+			jobController := NewJobController(channelRepo, consumerRepo, deliveryJobRepo)
+			testRouter := createTestRouter(jobController)
+			testURI := jobController.FormatAsRelativeLink(
+				httprouter.Param{Key: channelIDPathParamKey, Value: jobTestChannel.ChannelID},
+				httprouter.Param{Key: consumerIDPathParamKey, Value: jobTestPullConsumer.ConsumerID},
+				httprouter.Param{Key: jobIDPathParamKey, Value: job.ID.String()},
+			)
+			t.Log(testURI)
+
+			bodyString := "{\"NextState\": \"" + test.next.String() + "\",\"IncrementalTimeout\":" + strconv.Itoa(int(test.timeout)) + "}"
+			requestBody := ioutil.NopCloser(strings.NewReader(bodyString))
+			req, err := http.NewRequest(http.MethodPost, testURI, requestBody)
+			assert.NoError(t, err)
+
+			req.Header.Add(headerContentType, jobPostTestContentType)
+			req.Header.Add(headerChannelToken, jobTestChannel.Token)
+			req.Header.Add(headerConsumerToken, jobTestPullConsumer.Token)
+
+			rr := httptest.NewRecorder()
+			testRouter.ServeHTTP(rr, req)
+			assert.Equal(t, http.StatusAccepted, rr.Code)
+
+			updatedJob, err := deliveryJobRepo.GetByID(job.ID.String())
+			assert.NoError(t, err)
+			assert.Equal(t, test.next, updatedJob.Status)
+			assert.Equal(t, test.timeout, updatedJob.IncrementalTimeout)
+		})
+	}
+}
+
+func TestJobControllerPost_TimeoutValid(t *testing.T) {
+	t.Parallel()
+	job, err := setupTestJob()
+	assert.NoError(t, err)
+
+	jobController := NewJobController(channelRepo, consumerRepo, deliveryJobRepo)
+	testRouter := createTestRouter(jobController)
+	testURI := jobController.FormatAsRelativeLink(
+		httprouter.Param{Key: channelIDPathParamKey, Value: jobTestChannel.ChannelID},
+		httprouter.Param{Key: consumerIDPathParamKey, Value: jobTestPullConsumer.ConsumerID},
+		httprouter.Param{Key: jobIDPathParamKey, Value: job.ID.String()},
+	)
+	t.Log(testURI)
+	expected_timeout := uint(10)
+	tests := []struct {
+		current data.JobStatus
+		next    data.JobStatus
+		timeout time.Duration
+	}{
+		{current: data.JobQueued, next: data.JobInflight, timeout: 10},
+		{current: data.JobInflight, next: data.JobInflight},
+		{current: data.JobInflight, next: data.JobDead},
+		{current: data.JobDead, next: data.JobInflight, timeout: 10},
```

Comment:
Do we want the timeout to be updated when provided with Dead to Inflight transition? That doesn't happen now. This update should be made in the `MarkDeadJobAsInflight` method.
Using a different timeout value like `15` could have caught this.